src/data/finset/basic.lean

```diff
@@ -1304,6 +1304,12 @@ theorem range_mono : monotone range := λ _ _, range_subset.2
 lemma mem_range_succ_iff {a b : ℕ} : a ∈ finset.range b.succ ↔ a ≤ b :=
 finset.mem_range.trans nat.lt_succ_iff
 
+lemma mem_range_le {n x : ℕ} (hx : x ∈ range n) : x ≤ n :=
+(mem_range.1 hx).le
+
+lemma mem_range_sub_ne_zero {n x : ℕ} (hx : x ∈ range n) : n - x ≠ 0 :=
+ne_of_gt $ nat.sub_pos_of_lt $ mem_range.1 hx
```

Comment:
```suggestion
(nat.sub_pos_of_lt $ mem_range.1).ne
```
probably works

Reply:
Gives an error : 

`invalid field notation, 'ne' is not a valid "field" because environment does not contain 'gt.ne'`